tests/test_evaluators/test_ClassificationEvaluator.py

```diff
@@ -0,0 +1,569 @@
+        scores_with_cache, test_cache_after_cache_usage = eval_binary_cache_test(
+            model_pytorch, test_cache=test_cache_initial
+        )
+
+        assert torch.equal(test_cache_initial, test_cache_after_cache_usage)
+        for metric_key in scores_with_cache.keys():
+            if "accuracy" in metric_key or "f1" in metric_key or "ap" in metric_key:
+                assert 0 <= scores_with_cache[metric_key] <= 1
+
+    @pytest.mark.parametrize(
+        "train_sentences, train_labels, test_sentences, test_labels, task_name, limit, expected_train_len, expected_test_len, is_binary_after_limit, expected_exception",
+        [
+            (
+                SENTENCES_TRAIN_BINARY,
+                Y_TRAIN_BINARY,
+                SENTENCES_TEST_BINARY,
+                Y_TEST_BINARY,
+                "test_knn_pytorch_limit_binary",
+                3,
+                3,
+                2,
+                True,
+                None,
+            ),
+            (
+                SENTENCES_TRAIN_BINARY,
+                Y_TRAIN_BINARY,
+                SENTENCES_TEST_BINARY,
+                Y_TEST_BINARY,
+                "test_knn_pytorch_limit_not_binary",
+                1,
+                1,
+                1,
+                False,
+                None,
+            ),
+        ],
+    )
+    def test_limit_parameter(
+        self,
+        model_pytorch,
+        train_sentences,
+        train_labels,
+        test_sentences,
+        test_labels,
+        task_name,
+        limit,
+        expected_train_len,
+        expected_test_len,
+        is_binary_after_limit,
+        expected_exception,
+    ):
+        if expected_exception:
+            with pytest.raises(expected_exception):
+                eval_limited = kNNClassificationEvaluatorPytorch(
+                    train_sentences,
+                    train_labels,
+                    test_sentences,
+                    test_labels,
+                    task_name=task_name,
+                    limit=limit,
+                )
+                eval_limited(model_pytorch)
+        else:
+            eval_limited = kNNClassificationEvaluatorPytorch(
+                train_sentences,
+                train_labels,
+                test_sentences,
+                test_labels,
+                task_name=task_name,
+                limit=limit,
+            )
+            assert len(eval_limited.sentences_train) == expected_train_len
+            assert len(eval_limited.y_train) == expected_train_len
+            assert len(eval_limited.sentences_test) == expected_test_len
+            assert len(eval_limited.y_test) == expected_test_len
+
+            scores, _ = eval_limited(model_pytorch)
+            assert "accuracy" in scores
+            assert "f1" in scores
+            if is_binary_after_limit:
+                assert "ap" in scores
+            else:
+                assert "ap" not in scores
+
+
+class TestLogRegClassificationEvaluator:
```

Comment:
I don't think we need the class here. It also leads to duplication across the three different evaluators. Since we in v2 wil reduce the ClassificationEvalutors to just one class (see comment above), let us just focus on just logreg for now, feel free to delete the rest.